### ISSUE TITLE
Basic/simple example of showing a clickable text label in the middle of a link?

### ISSUE THREAD

**Issue by ken-murray** [vasturiano/force-graph#213]
Using ForceGraph2D in a React app, can anyone provide a basic/simple example of showing a clickable text label in the middle of a link between two nodes?  e.g.

<ForceGraph2D
	 ...

	 linkPositionUpdate={(link, { start, end }) => {
		const textPos = Object.assign({},...['x', 'y'].map(c => ({
			[c]: start[c] + (end[c] - start[c]) / 2 		// calc middle point
		})));
						
		...	
	}}
>

Thanks.

**Comment by vasturiano** [comment]
@ken-murray you can use `linkPointerAreaPaint` to define custom clickable areas of the links - by painting it with the provided color.

Also, if your question is React specific, feel free to post it on this repo: https://github.com/vasturiano/react-force-graph

**Comment by ken-murray** [comment]
Thanks again.  https://vasturiano.github.io/react-force-graph/example/text-links/index-3d.html is exactly what I need, except using ForceGraph2D instead.  I copied the relevant code, but I'm guessing SpriteText doesn't work for ForceGraph2D?

**Comment by ken-murray** [comment]
So, finally got the link labels showing - a bit misplaced and the rendering animation is very slow now.  Guessing this gets called on every frame:

	linkCanvasObjectMode={() => 'after'}
	linkCanvasObject={(link, ctx, globalScale) => {
		//console.log(link);
		let fontSize = 12;
		ctx.font = `${fontSize}px "Orbitron,sans-serif"`;
		//console.log("link label="+ link.label);
		const textWidth = ctx.measureText(link.label).width;
		let bckgDimensions = [textWidth, fontSize].map(n => n + fontSize * 0.2);
		//console.log(link);
		ctx.textAlign = 'center';
		ctx.textBaseline = 'middle';
		ctx.fillStyle = 'rgba(100, 0, 100, 1)';
		const midX = link.source.x + (link.source.x - link.target.x)/2;
		const midY = link.source.y + (link.source.y - link.target.y)/2;
		ctx.fillText(link.label, midX - textWidth/2, midY);
		link.__bckgDimensions = bckgDimensions;
	}}

Any suggestions from anyone on improving this would be much appreciated.

**Comment by vasturiano** [comment]
@ken-murray yes this function needs to be ran at every frame to ensure the canvas keeps reacting to changes. So, any computation within this method that can be cached/optimized always helps with a performance.

Have you profiled this function to try to find out which set of instructions are causing the biggest performance hit?
Or, if you can make a reproducible example on [codepen](codepen.io) we can also try to debug that way.

**Comment by ken-murray** [comment]
Thanks, just found your example here: https://vasturiano.github.io/force-graph/example/text-links/
I'll change my code accordingly and see if I can get my labels closer to where they should be first.
In terms of speed, it would be better if we could just do all this in the linkPositionUpdate() function.  Maybe I can put these labels into a list in the first tick/frame/component mount and then translate/move them in the linkPositionUpdate() function.  No rush for this, so I'll spend some time thinking about it.

Thanks again for your wonderful library and for all your help :-)

**Comment by ken-murray** [comment]
Copied your code and it works perfectly.  If I ever get funding for my hobby project I'll send you a good portion of it.  Thank you :-)

![Graph 2D Link Labels](https://user-images.githubusercontent.com/47371187/127725175-b8014138-86b3-4f5f-b163-20023021dcd3.png)
